python_modules/libraries/dagster-dbt/dagster_dbt/core/dbt_cli_event.py

```diff
@@ -432,12 +433,22 @@ def to_default_asset_events(
             asset_check_key = get_asset_check_key_for_test(
                 manifest, dagster_dbt_translator, test_unique_id=unique_id
             )
+            if not context or not context.has_assets_def:
+                selected_check_keys = set()
+            else:
+                selected_check_keys = {
+                    *context.selected_asset_check_keys,
+                    *get_checks_on_sources_upstream_of_selected_assets(
+                        assets_def=context.assets_def,
+                        selected_asset_keys=context.selected_asset_keys,
+                    ),
+                }
 
             if (
                 context
                 and has_asset_def
                 and asset_check_key is not None
-                and asset_check_key in context.selected_asset_check_keys
+                and asset_check_key in selected_check_keys
```

Comment:
one question -- now that we've loosened up the requirements on what asset checks are "allowed" to be emitted at runtime, could we just remove this clause entirely? e.g. we just make the logic "if the asset check key exists, then we'll emit an AssetCheckResult? Alternatively, could just check that the asset check key is within the keys found in node_check_specs_by_output_name (i.e. it's a valid check key, we just don't care if that happens to be associated with a source or not).

happy to be told there's a good reason that this is undesirable, just wanted to double check that we can't get away with the easier solution.

Reply:
I believe you; I also want to separate out any potential behavior changes to existing dbt stuff from this PR